ui/ptdockwidget.py

```diff
@@ -316,6 +316,8 @@ def addLayer(self, layer1=None):
 
         self.tableViewTool.addLayer(self.iface, self.mdl, layer1)
         self.profiletoolcore.updateProfil(self.profiletoolcore.pointstoDraw, False)
+        if layer1 is None: # no layer selected in the dropdown
+            return
```

Comment:
I would move this before `self.tableViewTool.addLayer(self.iface, self.mdl, layer1)`
If layer1 is none, adding it to tableview is pointless.

Reply:
That's what I thought when I saw the code, but doing that prevents the "layer selector" dialog from opening.

Reply:
Lot of spaghetti code in here, let's merge this so